test/lib/commands/adduser.js

```diff
@@ -1,190 +1,176 @@
+const MockRegistry = require('../../fixtures/mock-registry.js')
+const stream = require('stream')
 
 t.test('usage', async t => {
+  const { npm } = await loadMockNpm(t)
+  const adduser = await npm.cmd('adduser')
   t.match(adduser.usage, 'adduser', 'usage has command name in it')
 })
 
 t.test('simple login', async t => {
-  await adduser.exec([])
-  t.equal(
-    registryOutput,
-    'Log in on https://registry.npmjs.org/',
-    'should have correct message result'
-  )
-
-  t.same(
-    deletedConfig,
-    {
-      _token: 'user',
-      _password: 'user',
-      username: 'user',
-      _auth: 'user',
-      _authtoken: 'user',
-      '-authtoken': 'user',
-      _authToken: 'user',
-      '//registry.npmjs.org/:-authtoken': 'user',
-      '//registry.npmjs.org/:_authToken': 'user',
-      '//registry.npmjs.org/:_authtoken': 'user',
-      '//registry.npmjs.org/:always-auth': 'user',
-      '//registry.npmjs.org/:email': 'user',
+  const stdin = new stream.PassThrough()
+  stdin.write('test-user\n')
+  stdin.write('test-password\n')
+  stdin.write('test-email@npmjs.org\n')
+  mockGlobals(t, {
+    'process.stdin': stdin,
+    // 'process.stdout': new stream.PassThrough(), // to quiet readline
+  }, { replace: true })
+  const { npm, home } = await loadMockNpm(t, {
+    homeDir: {
+      // These all get cleaned up by config.setCredentialsByURI
+      '.npmrc': [
+        '_token=user',
+        '_password=user',
+        'username=user',
+        '_auth=user',
+        '_authtoken=user',
+        '-authtoken=user',
+        '_authToken=user',
+        '//registry.npmjs.org/:-authtoken=user',
+        '//registry.npmjs.org/:_authToken=user',
+        '//registry.npmjs.org/:_authtoken=user',
+        '//registry.npmjs.org/:always-auth=user',
+        '//registry.npmjs.org/:email=test-email-old@npmjs.org',
+      ].join('\n'),
     },
-    'should delete token in user config'
-  )
-
+  })
+  const registry = new MockRegistry({
+    tap: t,
+    registry: npm.config.get('registry'),
+  })
+  registry.couchlogin({
+    username: 'test-user',
+    password: 'test-password',
+    email: 'test-email@npmjs.org',
+    token: 'npm_test-token',
+  })
+  await npm.exec('adduser', [])
+  t.same(npm.config.get('email'), 'test-email-old@npmjs.org')
+  t.same(npm.config.get('//registry.npmjs.org/:_authToken'), 'npm_test-token')
+  const rc = fs.readFileSync(path.join(home, '.npmrc'), 'utf8')
   t.same(
-    setConfig,
-    {
-      '//registry.npmjs.org/:_password': { value: 'cA==', where: 'user' },
-      '//registry.npmjs.org/:username': { value: 'u', where: 'user' },
-      email: { value: 'u@npmjs.org', where: 'user' },
-    },
-    'should set expected user configs'
-  )
-
-  t.equal(
-    result,
-    'success',
-    'should output auth success msg'
+    rc.trim().split(os.EOL), [
+      '//registry.npmjs.org/:_authToken=npm_test-token',
+      'email=test-email-old@npmjs.org',
+    ], 'should only have token and un-nerfed old email'
```

Comment:
Is this the incorrect behavior? Would be nice to flag it explicitly for future removal/etc

Reply:
This is an `@npmcli/config` bug, and one I was hoping was on your radar for the deprecated config warnings.